mage_ai/data_loader/postgres.py

```diff
@@ -0,0 +1,54 @@
+from mage_ai.data_loader.base import BaseSQL
+from pandas import DataFrame, read_sql
+from psycopg2 import connect
+from typing import Mapping, Sequence, Union
+
+
+class Postgres(BaseSQL):
+    """
+    Loads data from a PostgreSQL database.
+    """
+
+    def __init__(self, **kwargs) -> None:
+        """
+        Initializes the data loader. Below are some sample arguments, for the full list see libpq parameter keywords.
+
+        Args:
+            dbname (str): The name of the database to connect to.
+            user (str): The user with which to connect to the database with.
+            password (str): The login password for the user.
+            host (str): Path to host address for database.
+            port (str): Port on which the database is running.
+        """
+        super().__init__(**kwargs)
+
+    def open(self) -> None:
+        """
+        Opens a connection to the PostgreSQL database specified by the parameters.
+        """
+        self._ctx = connect(**self.settings)
+
+    def query(self, query_string: str, query_vars: Union[Sequence, Mapping] = None) -> None:
```

Comment:
Are you going to use this method somewhere? or is this a method allowing user to use?

Reply:
This is a method allowing the user to send queries to the dataset that don't return any data. This could be, for example, joining tables or adding new entries before the table is imported to a database using `load`.

This could be done manually using the `self.conn` property, but to abstract the connection away I provide this method.